sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,75 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
 {
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  std::string name;
+  std::string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(std::string name, std::string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  std::string getName();
```

Comment:
If these two functions just return member variable `name` and `value`, would it be better to return by const reference?

Reply:
yeah, I will update many places to use const refs